src/main/java/org/broadinstitute/hellbender/tools/copynumber/datacollection/PerBaseCountCollector.java

```diff
@@ -0,0 +1,150 @@
+                final int pairBaseQuality = unpairedTemplateQuality.get(queryName);
+                final Nucleotide pairBase = unpairedTemplateBase.get(queryName);
+                if (pairBaseQuality > baseQuality) {
+                    // If the other read of the pair has a higher base quality we count its base
+                    final int count = perBaseCount.get(pairBase) + 1;
+                    perBaseCount.put(pairBase, count);
+                } else if(pairBaseQuality == baseQuality) {
+                    // If the paired bases have equal quality, count the base from the first of pair read
+                    if(firstOfPair) {
+                        final int count = perBaseCount.get(base) + 1;
+                        perBaseCount.put(base, count);
+                    } else {
+                        final int count = perBaseCount.get(pairBase) + 1;
+                        perBaseCount.put(pairBase, count);
+                    }
+                } else if(baseQuality > pairBaseQuality) {
+                    // Otherwise we count this reads base
+                    final int count = perBaseCount.get(base) + 1;
+                    perBaseCount.put(base, count);
+                }
+                // We've found the read pair, so remove the first encountered read from the unpaired HashMaps
+                unpairedTemplateBase.remove(queryName);
+                unpairedTemplateQuality.remove(queryName);
+            } else {
+                // This is the first read of the pair, so add it to the unpaired HashMaps
+                unpairedTemplateBase.put(queryName, base);
+                unpairedTemplateQuality.put(queryName, baseQuality);
+            }
+        }
+
+        // add all the unpaired bases to the perBaseCount
+        for(final Nucleotide base : unpairedTemplateBase.values()) {
+            final int count = perBaseCount.get(base) + 1;
+            perBaseCount.put(base, count);
+        }
+
+        perBaseCounts.add(new PerBaseCount(
+                new SimpleInterval(locus.getContig(), locus.getStart(), locus.getEnd()),
+                perBaseCount));
+    }
+
+    /**
+     * Get the per-base counts gathered so far.
+     *
+     * @return a <em>reference</em> to the PerBaseCountCollection
+     */
+    public PerBaseCountCollection getPerBaseCounts() {
+        return new PerBaseCountCollection(metadata, perBaseCounts);
+    }
+
+    /**
+     * Reminder that any additional information used through this method will not be able to enforce the minBaseQuality.
+     *
+     * @param perBaseCountCollector input data to combine with this.
+     */
+    public void collectFromCollector(final PerBaseCountCollector perBaseCountCollector) {
+        if (perBaseCountCollector != null) {
+            this.perBaseCounts.addAll(perBaseCountCollector.getPerBaseCounts().getRecords());
+        }
+    }
+
+    /** TODO: Docs and input parameter checking
```

Comment:
We should clean up this doc here and in `AllelicCountCollector` (probably fine to just remove it).